README.md

```diff
@@ -56,7 +57,178 @@ Builds the app for production to the build folder.
 
 @TODO: Add production build instructions.
 
-
 ## Event Model
 
 ![Event model](docs/EventModel.png)
+
+## How to create a template (WIP - perhaps move this section to another place)
+
+The templates are created with react ([remote component](https://github.com/Paciolan/remote-component)). The components that are loaded in the client should be compiled.
+TODO: write more about remote component when implemented
+
+### Obligatory
+
+The client loads a compiled react component.
+A component will be created like below
+
+```html
+<ExampleComponent
+  slide={slide}
+  slideDone={slideDone}
+  content={slide.content}
+  run={run}
+/>
+```
+
+**slide**: The slide object.
+**slideDone**: A function that is called when the slide is done.
+**Content**: is the slide content.
+**Run**: is a boolean whether the slide is running or not.
+
+And the following code snippet should be in the component:
+
+```javascript
+/**
+ * Setup slide run function.
+ */
+const slideExecution = new BaseSlideExecution(slide, slideDone);
+useEffect(() => {
+  if (run) {
+    slideExecution.start(slide.duration);
+  } else {
+    slideExecution.stop();
+  }
+
+  return function cleanup() {
+    slideExecution.stop();
+  };
+}, [run]);
+```
+
+#### JSON for backend form
+
+If the template should be configured from the admin backend, it needs a json file. As of now, the form is:
```

Comment:
To populate the slide with data an admin form is needed. This is configured in a json file.